### PR TITLE
alfa-bench test

### PR DESCRIPTION
**Pull request by NiclasHedam**
This tests the fundamental logic behind the alfa-bench package.

### CODE REVIEW

**Comment by NiclasHedam** [comment]
I need to see some code on how you want to achieve this.
Currently, I import a clean benchmark.js instance and overwrite it's run-handler only to run once and return a predetermined result. This solution is the best I could come up with, while not doing a lot of nasty stuff.

File: packages/alfa-bench/src/benchmark.ts [hunk 1]
```diff
@@ -8,25 +8,69 @@ interface Target {
...
+ */
+export interface Event {
+  readonly target: Target;
+}
+
+/**
+ * @internal
+ */
+export interface Notifier {
+  out: (message: string) => void;
```
Comment (kasperisager) [inline packages/alfa-bench/src/benchmark.ts]: `out` -> `print`

File: packages/alfa-bench/src/benchmark.ts [hunk 2]
```diff
@@ -8,25 +8,69 @@ interface Target {
...
+ * @internal
+ */
+export interface Event {
+  readonly target: Target;
+}
+
+/**
+ * @internal
+ */
+export interface Notifier {
```
Comment (kasperisager) [inline packages/alfa-bench/src/benchmark.ts]: `Printer`?

File: packages/alfa-bench/src/benchmark.ts [hunk 3]
```diff
@@ -8,25 +8,69 @@ interface Target {
...
+  out: (message: string) => void;
+}
+
+const defaultNotifier: Notifier = {
+  out: message => {
+    process.stderr.write(`${message}\n`);
```
Comment (kasperisager) [inline packages/alfa-bench/src/benchmark.ts]: `stdout`

File: packages/alfa-bench/test/benchmark.spec.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      ]);
+    },
+    run: () => {},
+    add: (title: string, callback: () => void | Promise<void>) => suite
+  };
+
+  benchmark(suite, result, notifier);
+});
+
+test("Outputs stats in each cycle", t => {
```
Comment (kasperisager) [inline packages/alfa-bench/test/benchmark.spec.ts]: I'd remove this.

**Comment by NiclasHedam** [comment]
`printer.print(`${output}\n`);` is some really interesting code.

**Comment by NiclasHedam** [comment]
Now this PR fails because master is broken. :joy:

**Pull request merged**